## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#bd3854aa-7a4c-4d52-bc11-463564241a70) |
| mint-gateway-s3.sh | :x: [more...](#27611743-9da3-4ecf-b732-7cc32477061d) |
| mint-erasure.sh | :x: [more...](#c36fea85-a30f-4b53-82ee-bd776244310e) |
| mint-dist-erasure.sh | :x: [more...](#3986c541-e306-4ed7-9219-e9be257d33f0) |
| mint-zoned.sh | :x: [more...](#59898efe-312c-4784-a260-15c3561ebc87) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#6e6c11a6-ace6-4dd1-9f38-b63117a18437) |
<a name="59898efe-312c-4784-a260-15c3561ebc87"></a>
### 12333-c14b5dc/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31624
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f5702e2aad23:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 25 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 47 seconds
(8/15) Running minio-dotnet tests ... done in 48 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 19 seconds
(10/15) Running minio-java tests ... FAILED in 58 seconds
{
  "name": "minio-java",
  "function": "listObjects()",
  "args": "[bucket, prefix, recursive, 1050 objects]",
  "duration": 46618,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InternalError, message = We encountered an internal error, please try again.: cause(io: read/write on closed pipe), bucketName = minio-java-test-mbloj0, objectName = null, resource = /minio-java-test-mbloj0, requestId = 16809EC05E57354A, hostId = 7cb996e8-7dc0-4978-84fa-f7db8a04e425)\nrequest={method=GET, url=http://minio-c3.minio.io:31624/minio-java-test-mbloj0?list-type=2&prefix=minio&continuation-token=bWluaW8tamF2YS10ZXN0LXFxNGVtM1ttaW5pb19jYWNoZTp2MSxpZDphNTZiNmU0OC1iOWE5LTRhZmMtOGIyZi00NmM0ODFlMWM0Mjdd&max-keys=1000&encoding-type=url&delimiter=, headers=Host: minio-c3.minio.io:31624\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210520T002654Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210520/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=500, headers=Accept-Ranges: bytes\nContent-Length: 371\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16809EC05E57354A\nX-Xss-Protection: 1; mode=block\nDate: Thu, 20 May 2021 00:26:54 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.listObjectsV2(MinioClient.java:4368), io.minio.MinioClient$2$1.populateResult(MinioClient.java:2016), io.minio.MinioClient$ObjectIterator.populate(MinioClient.java:1882), io.minio.MinioClient$ObjectIterator.hasNext(MinioClient.java:1924), FunctionalTest.testListObjects(FunctionalTest.java:1398), FunctionalTest.listObjects(FunctionalTest.java:1455), FunctionalTest.runObjectTests(FunctionalTest.java:3747), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(10/15) Running minio-js tests ... done in 50 seconds
(11/15) Running minio-py tests ... FAILED in 2 minutes and 32 seconds
{
  "name": "minio-py:test_list_objects_with_1001_files",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-adbab374-2362-407b-a2a7-5ee373671282",
    "object_name": "8306a461-ae8c-43d7-bee7-2c3d1bce5970_0 ~ 8306a461-ae8c-43d7-bee7-2c3d1bce5970_1000"
  },
  "message": "HTTPConnectionPool(host='minio-c3.minio.io', port=31624): Max retries exceeded with url: /minio-py-test-adbab374-2362-407b-a2a7-5ee373671282?continuation-token=ODMwNmE0NjEtYWU4Yy00M2Q3LWJlZTctMmMzZDFiY2U1OTcwXzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZDk4OTkwZmQtZTg4Zi00NTQ1LTg3NzItMzQ3NjE4N2E0YzM3XQ%3D%3D&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=ODMwNmE0NjEtYWU4Yy00M2Q3LWJlZTctMmMzZDFiY2U1OTcwXzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZDk4OTkwZmQtZTg4Zi00NTQ1LTg3NzItMzQ3NjE4N2E0YzM3XQ%3D%3D (Caused by ResponseError('too many 500 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1243, in test_list_objects_with_1001_files\n    _test_list_objects_api(bucket_name, no_of_created_files)\n  File \"/mint/run/core/minio-py/tests.py\", line 1123, in _test_list_objects_api\n    for obj in objects:\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2758, in _list_objects\n    response = self._execute(\"GET\", bucket_name, query_params=query)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-c3.minio.io', port=31624): Max retries exceeded with url: /minio-py-test-adbab374-2362-407b-a2a7-5ee373671282?continuation-token=ODMwNmE0NjEtYWU4Yy00M2Q3LWJlZTctMmMzZDFiY2U1OTcwXzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZDk4OTkwZmQtZTg4Zi00NTQ1LTg3NzItMzQ3NjE4N2E0YzM3XQ%3D%3D&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=ODMwNmE0NjEtYWU4Yy00M2Q3LWJlZTctMmMzZDFiY2U1OTcwXzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZDk4OTkwZmQtZTg4Zi00NTQ1LTg3NzItMzQ3NjE4N2E0YzM3XQ%3D%3D (Caused by ResponseError('too many 500 error responses',))\n",
  "duration": 97797
}
(11/15) Running s3cmd tests ... done in 19 seconds
(12/15) Running s3select tests ... done in 8 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="bd3854aa-7a4c-4d52-bc11-463564241a70"></a>
### 12333-c14b5dc/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:31925
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 87519dedfcaf:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... FAILED in 23 seconds
{
  "name": "awscli",
  "duration": 11963,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "fatal error: An error occurred (InternalError) when calling the ListObjectsV2 operation (reached max retries: 4): We encountered an internal error, please try again.: cause(io: read/write on closed pipe)remove_bucket failed: Unable to delete all objects in the bucket, bucket will not be deleted."
}
(5/15) Running healthcheck tests ... done in 1 seconds
(6/15) Running mc tests ... done in 41 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 3 seconds
(8/15) Running minio-go tests ... done in 2 minutes and 26 seconds
(9/15) Running minio-java tests ... FAILED in 1 minutes and 17 seconds
{
  "name": "minio-java",
  "function": "listObjects()",
  "args": "[bucket, prefix, recursive, 1050 objects]",
  "duration": 55604,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InternalError, message = We encountered an internal error, please try again.: cause(io: read/write on closed pipe), bucketName = minio-java-test-2o4a0ip, objectName = null, resource = /minio-java-test-2o4a0ip, requestId = 16809EBA0CF5E488, hostId = a4fb3594-7cf6-437c-9685-ab7b548b1287)\nrequest={method=GET, url=http://minio-dev7.minio.io:31925/minio-java-test-2o4a0ip?list-type=2&prefix=minio&continuation-token=bWluaW8tamF2YS10ZXN0LXA5NTUzN1ttaW5pb19jYWNoZTp2MSxpZDo4YTlhYTcwNi1mODRjLTQ4YTAtOWI3NC0wMDI3N2Q1MGRkZWRd&max-keys=1000&encoding-type=url&delimiter=, headers=Host: minio-dev7.minio.io:31925\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210520T002627Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210520/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=500, headers=Accept-Ranges: bytes\nContent-Length: 373\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16809EBA0CF5E488\nX-Xss-Protection: 1; mode=block\nDate: Thu, 20 May 2021 00:26:27 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.listObjectsV2(MinioClient.java:4368), io.minio.MinioClient$2$1.populateResult(MinioClient.java:2016), io.minio.MinioClient$ObjectIterator.populate(MinioClient.java:1882), io.minio.MinioClient$ObjectIterator.hasNext(MinioClient.java:1924), FunctionalTest.testListObjects(FunctionalTest.java:1398), FunctionalTest.listObjects(FunctionalTest.java:1455), FunctionalTest.runObjectTests(FunctionalTest.java:3747), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(9/15) Running minio-js tests ... done in 49 seconds
(10/15) Running minio-py tests ... FAILED in 2 minutes and 27 seconds
{
  "name": "minio-py:test_list_objects_with_1001_files",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-0619f403-0816-417a-8616-930b6b0b48b0",
    "object_name": "360c763e-80fb-4dff-8be0-ae0249db1787_0 ~ 360c763e-80fb-4dff-8be0-ae0249db1787_1000"
  },
  "message": "HTTPConnectionPool(host='minio-dev7.minio.io', port=31925): Max retries exceeded with url: /minio-py-test-0619f403-0816-417a-8616-930b6b0b48b0?continuation-token=MzYwYzc2M2UtODBmYi00ZGZmLThiZTAtYWUwMjQ5ZGIxNzg3XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZGIwZTZmZjMtYTgxMy00NTFlLWE0NjctYzI5Y2RjODM2ZTgwXQ%3D%3D&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=MzYwYzc2M2UtODBmYi00ZGZmLThiZTAtYWUwMjQ5ZGIxNzg3XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZGIwZTZmZjMtYTgxMy00NTFlLWE0NjctYzI5Y2RjODM2ZTgwXQ%3D%3D (Caused by ResponseError('too many 500 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1243, in test_list_objects_with_1001_files\n    _test_list_objects_api(bucket_name, no_of_created_files)\n  File \"/mint/run/core/minio-py/tests.py\", line 1123, in _test_list_objects_api\n    for obj in objects:\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2758, in _list_objects\n    response = self._execute(\"GET\", bucket_name, query_params=query)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev7.minio.io', port=31925): Max retries exceeded with url: /minio-py-test-0619f403-0816-417a-8616-930b6b0b48b0?continuation-token=MzYwYzc2M2UtODBmYi00ZGZmLThiZTAtYWUwMjQ5ZGIxNzg3XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZGIwZTZmZjMtYTgxMy00NTFlLWE0NjctYzI5Y2RjODM2ZTgwXQ%3D%3D&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=MzYwYzc2M2UtODBmYi00ZGZmLThiZTAtYWUwMjQ5ZGIxNzg3XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZGIwZTZmZjMtYTgxMy00NTFlLWE0NjctYzI5Y2RjODM2ZTgwXQ%3D%3D (Caused by ResponseError('too many 500 error responses',))\n",
  "duration": 92212
}
(10/15) Running s3cmd tests ... done in 17 seconds
(11/15) Running s3select tests ... done in 5 seconds
(12/15) Running security tests ... done in 0 seconds

Executed 12 out of 15 tests successfully.
```
<a name="3986c541-e306-4ed7-9219-e9be257d33f0"></a>
### 12333-c14b5dc/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31107
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f9e6961e2ac8:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 2 minutes and 22 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 57 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 3 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 43 seconds
(10/15) Running minio-java tests ... FAILED in 39 seconds
{
  "name": "minio-java",
  "function": "listObjects()",
  "args": "[bucket, prefix, recursive, 1050 objects]",
  "duration": 29354,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InternalError, message = We encountered an internal error, please try again.: cause(io: read/write on closed pipe), bucketName = minio-java-test-2kcndfg, objectName = null, resource = /minio-java-test-2kcndfg, requestId = 16809EC76982911C, hostId = 127ada83-33e4-4a5a-a3e3-2dcb80e202e4)\nrequest={method=GET, url=http://minio-c3.minio.io:31107/minio-java-test-2kcndfg?list-type=2&prefix=minio&continuation-token=bWluaW8tamF2YS10ZXN0LXEwYjI4b1ttaW5pb19jYWNoZTp2MSxpZDphNjRhMTI3My1jMjAyLTQ0YzgtODU4NS01NmVjZTU0OTQzOGZd&max-keys=1000&encoding-type=url&delimiter=, headers=Host: minio-c3.minio.io:31107\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210520T002725Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210520/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=500, headers=Accept-Ranges: bytes\nContent-Length: 373\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16809EC76982911C\nX-Xss-Protection: 1; mode=block\nDate: Thu, 20 May 2021 00:27:25 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.listObjectsV2(MinioClient.java:4368), io.minio.MinioClient$2$1.populateResult(MinioClient.java:2016), io.minio.MinioClient$ObjectIterator.populate(MinioClient.java:1882), io.minio.MinioClient$ObjectIterator.hasNext(MinioClient.java:1924), FunctionalTest.testListObjects(FunctionalTest.java:1398), FunctionalTest.listObjects(FunctionalTest.java:1455), FunctionalTest.runObjectTests(FunctionalTest.java:3747), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(10/15) Running minio-js tests ... done in 46 seconds
(11/15) Running minio-py tests ... FAILED in 2 minutes and 17 seconds
{
  "name": "minio-py:test_list_objects_with_1001_files",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-f242b215-7ba6-4aa8-b7ea-4f5cd0369d5a",
    "object_name": "515c6814-7807-4d48-bad1-5bff6e0b6799_0 ~ 515c6814-7807-4d48-bad1-5bff6e0b6799_1000"
  },
  "message": "HTTPConnectionPool(host='minio-c3.minio.io', port=31107): Max retries exceeded with url: /minio-py-test-f242b215-7ba6-4aa8-b7ea-4f5cd0369d5a?continuation-token=NTE1YzY4MTQtNzgwNy00ZDQ4LWJhZDEtNWJmZjZlMGI2Nzk5XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6OTRkMjc5ODEtMzJmOS00OTUwLWE4NWMtZGIwNWYxNGRjZDU5XQ%3D%3D&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=NTE1YzY4MTQtNzgwNy00ZDQ4LWJhZDEtNWJmZjZlMGI2Nzk5XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6OTRkMjc5ODEtMzJmOS00OTUwLWE4NWMtZGIwNWYxNGRjZDU5XQ%3D%3D (Caused by ResponseError('too many 500 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1243, in test_list_objects_with_1001_files\n    _test_list_objects_api(bucket_name, no_of_created_files)\n  File \"/mint/run/core/minio-py/tests.py\", line 1123, in _test_list_objects_api\n    for obj in objects:\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2758, in _list_objects\n    response = self._execute(\"GET\", bucket_name, query_params=query)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-c3.minio.io', port=31107): Max retries exceeded with url: /minio-py-test-f242b215-7ba6-4aa8-b7ea-4f5cd0369d5a?continuation-token=NTE1YzY4MTQtNzgwNy00ZDQ4LWJhZDEtNWJmZjZlMGI2Nzk5XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6OTRkMjc5ODEtMzJmOS00OTUwLWE4NWMtZGIwNWYxNGRjZDU5XQ%3D%3D&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=NTE1YzY4MTQtNzgwNy00ZDQ4LWJhZDEtNWJmZjZlMGI2Nzk5XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6OTRkMjc5ODEtMzJmOS00OTUwLWE4NWMtZGIwNWYxNGRjZDU5XQ%3D%3D (Caused by ResponseError('too many 500 error responses',))\n",
  "duration": 83067
}
(11/15) Running s3cmd tests ... done in 19 seconds
(12/15) Running s3select tests ... done in 5 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="c36fea85-a30f-4b53-82ee-bd776244310e"></a>
### 12333-c14b5dc/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:30658
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5aa7e5420359:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 17 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 58 seconds
(8/15) Running minio-dotnet tests ... done in 35 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 13 seconds
(10/15) Running minio-java tests ... FAILED in 17 seconds
{
  "name": "minio-java",
  "function": "listObjects()",
  "args": "[bucket, prefix, recursive, 1050 objects]",
  "duration": 13289,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InternalError, message = We encountered an internal error, please try again.: cause(io: read/write on closed pipe), bucketName = minio-java-test-e320t9, objectName = null, resource = /minio-java-test-e320t9, requestId = 16809EB3B574E8E3, hostId = 43a13347-cc1c-4a9e-9ba3-3217237f661c)\nrequest={method=GET, url=http://minio-dev4.minio.io:30658/minio-java-test-e320t9?list-type=2&prefix=minio&continuation-token=bWluaW8tamF2YS10ZXN0LW9tb3RuOFttaW5pb19jYWNoZTp2MSxpZDpkYjNmMmQwMy05MmU2LTQyMTItOWM1YS0yMDFjMGIwYWJlOTdd&max-keys=1000&encoding-type=url&delimiter=, headers=Host: minio-dev4.minio.io:30658\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210520T002600Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210520/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=500, headers=Accept-Ranges: bytes\nContent-Length: 371\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16809EB3B574E8E3\nX-Xss-Protection: 1; mode=block\nDate: Thu, 20 May 2021 00:26:00 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.listObjectsV2(MinioClient.java:4368), io.minio.MinioClient$2$1.populateResult(MinioClient.java:2016), io.minio.MinioClient$ObjectIterator.populate(MinioClient.java:1882), io.minio.MinioClient$ObjectIterator.hasNext(MinioClient.java:1924), FunctionalTest.testListObjects(FunctionalTest.java:1398), FunctionalTest.listObjects(FunctionalTest.java:1455), FunctionalTest.runObjectTests(FunctionalTest.java:3747), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(10/15) Running minio-js tests ... done in 51 seconds
(11/15) Running minio-py tests ... FAILED in 1 minutes and 23 seconds
{
  "name": "minio-py:test_list_objects_with_1001_files",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-0314e1ab-dc13-4f60-a47f-c2aa4d99c50f",
    "object_name": "e74bcefe-0644-470c-8843-eebb1f1b5545_0 ~ e74bcefe-0644-470c-8843-eebb1f1b5545_1000"
  },
  "message": "HTTPConnectionPool(host='minio-dev4.minio.io', port=30658): Max retries exceeded with url: /minio-py-test-0314e1ab-dc13-4f60-a47f-c2aa4d99c50f?continuation-token=ZTc0YmNlZmUtMDY0NC00NzBjLTg4NDMtZWViYjFmMWI1NTQ1XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZTM1YzRkM2UtNWUyYi00Mzk5LTk3ZDYtOTJlZWE0MTEyMjBjXQ%3D%3D&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=ZTc0YmNlZmUtMDY0NC00NzBjLTg4NDMtZWViYjFmMWI1NTQ1XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZTM1YzRkM2UtNWUyYi00Mzk5LTk3ZDYtOTJlZWE0MTEyMjBjXQ%3D%3D (Caused by ResponseError('too many 500 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1243, in test_list_objects_with_1001_files\n    _test_list_objects_api(bucket_name, no_of_created_files)\n  File \"/mint/run/core/minio-py/tests.py\", line 1123, in _test_list_objects_api\n    for obj in objects:\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2758, in _list_objects\n    response = self._execute(\"GET\", bucket_name, query_params=query)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev4.minio.io', port=30658): Max retries exceeded with url: /minio-py-test-0314e1ab-dc13-4f60-a47f-c2aa4d99c50f?continuation-token=ZTc0YmNlZmUtMDY0NC00NzBjLTg4NDMtZWViYjFmMWI1NTQ1XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZTM1YzRkM2UtNWUyYi00Mzk5LTk3ZDYtOTJlZWE0MTEyMjBjXQ%3D%3D&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=ZTc0YmNlZmUtMDY0NC00NzBjLTg4NDMtZWViYjFmMWI1NTQ1XzE4OThbbWluaW9fY2FjaGU6djEsaWQ6ZTM1YzRkM2UtNWUyYi00Mzk5LTk3ZDYtOTJlZWE0MTEyMjBjXQ%3D%3D (Caused by ResponseError('too many 500 error responses',))\n",
  "duration": 32078
}
(11/15) Running s3cmd tests ... done in 20 seconds
(12/15) Running s3select tests ... done in 4 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="6e6c11a6-ace6-4dd1-9f38-b63117a18437"></a>
### 12333-c14b5dc/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:32233
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp cc65d0f08f68:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 34 seconds
{
  "name": "awscli",
  "duration": 2899,
  "function": "aws --endpoint-url http://minio-dev6.minio.io:32233 s3api copy-object --bucket awscli-mint-test-bucket-30464 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-30464/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 51 seconds
(7/15) Running minio-dotnet tests ... done in 54 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 25 seconds
{
  "args": {},
  "duration": 794,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 32 seconds
{
  "name": "minio-java",
  "function": "listObjects()",
  "args": "[bucket, prefix, recursive, 1050 objects]",
  "duration": 75068,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InternalError, message = We encountered an internal error, please try again.: cause(io: read/write on closed pipe), bucketName = minio-java-test-1rmn22j, objectName = null, resource = /minio-java-test-1rmn22j, requestId = 16809EC09F1A0792, hostId = eb490d90-516f-4227-897d-9ba23c75ecd1)\nrequest={method=GET, url=http://minio-dev6.minio.io:32233/minio-java-test-1rmn22j?list-type=2&prefix=minio&continuation-token=bWluaW8tamF2YS10ZXN0LW9zNm0wb1ttaW5pb19jYWNoZTp2MSxpZDo1YzBlOGJhYy1lMTQ0LTRkYjQtYWM4NC05ZjQyNGI4NDQwYTld&max-keys=1000&encoding-type=url&delimiter=, headers=Host: minio-dev6.minio.io:32233\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210520T002655Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210520/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=500, headers=Accept-Ranges: bytes\nContent-Length: 373\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16809EC09F1A0792\nX-Xss-Protection: 1; mode=block\nDate: Thu, 20 May 2021 00:26:55 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.listObjectsV2(MinioClient.java:4368), io.minio.MinioClient$2$1.populateResult(MinioClient.java:2016), io.minio.MinioClient$ObjectIterator.populate(MinioClient.java:1882), io.minio.MinioClient$ObjectIterator.hasNext(MinioClient.java:1924), FunctionalTest.testListObjects(FunctionalTest.java:1398), FunctionalTest.listObjects(FunctionalTest.java:1455), FunctionalTest.runObjectTests(FunctionalTest.java:3747), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 52 seconds
(9/15) Running minio-py tests ... FAILED in 2 minutes and 50 seconds
{
  "name": "minio-py:test_list_objects_with_1001_files",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-1da07746-2fb7-469c-8ec3-3794de758a42",
    "object_name": "b520c6e2-990a-431a-b6e4-2a57495dbf13_0 ~ b520c6e2-990a-431a-b6e4-2a57495dbf13_1000"
  },
  "message": "HTTPConnectionPool(host='minio-dev6.minio.io', port=32233): Max retries exceeded with url: /minio-py-test-1da07746-2fb7-469c-8ec3-3794de758a42?continuation-token=YjUyMGM2ZTItOTkwYS00MzFhLWI2ZTQtMmE1NzQ5NWRiZjEzXzE4OThbbWluaW9fY2FjaGU6djEsaWQ6YTJkMzZiYzctZjlmMS00NWJiLWE0ZGUtNjFkNTg3MjVlNjhlXQ%3D%3D&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=YjUyMGM2ZTItOTkwYS00MzFhLWI2ZTQtMmE1NzQ5NWRiZjEzXzE4OThbbWluaW9fY2FjaGU6djEsaWQ6YTJkMzZiYzctZjlmMS00NWJiLWE0ZGUtNjFkNTg3MjVlNjhlXQ%3D%3D (Caused by ResponseError('too many 500 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1243, in test_list_objects_with_1001_files\n    _test_list_objects_api(bucket_name, no_of_created_files)\n  File \"/mint/run/core/minio-py/tests.py\", line 1123, in _test_list_objects_api\n    for obj in objects:\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2758, in _list_objects\n    response = self._execute(\"GET\", bucket_name, query_params=query)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev6.minio.io', port=32233): Max retries exceeded with url: /minio-py-test-1da07746-2fb7-469c-8ec3-3794de758a42?continuation-token=YjUyMGM2ZTItOTkwYS00MzFhLWI2ZTQtMmE1NzQ5NWRiZjEzXzE4OThbbWluaW9fY2FjaGU6djEsaWQ6YTJkMzZiYzctZjlmMS00NWJiLWE0ZGUtNjFkNTg3MjVlNjhlXQ%3D%3D&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=YjUyMGM2ZTItOTkwYS00MzFhLWI2ZTQtMmE1NzQ5NWRiZjEzXzE4OThbbWluaW9fY2FjaGU6djEsaWQ6YTJkMzZiYzctZjlmMS00NWJiLWE0ZGUtNjFkNTg3MjVlNjhlXQ%3D%3D (Caused by ResponseError('too many 500 error responses',))\n",
  "duration": 114279
}
(9/15) Running s3cmd tests ... FAILED in 6 seconds
{
  "name": "s3cmd",
  "duration": "3238",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-7407/s3cmd-test-object-6305 92439c3f-87c9-4fb0-9705-e6c9270079cc\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 92439c3f-87c9-4fb0-9705-e6c9270079cc put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(9/15) Running s3select tests ... done in 6 seconds
(10/15) Running security tests ... done in 0 seconds

Executed 10 out of 15 tests successfully.
```
<a name="27611743-9da3-4ecf-b732-7cc32477061d"></a>
### 12333-c14b5dc/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:32102
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d7a9e7d45133:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 1 minutes and 16 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 45 seconds
(8/15) Running minio-dotnet tests ... done in 33 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 4 seconds
(10/15) Running minio-java tests ... FAILED in 33 seconds
{
  "name": "minio-java",
  "function": "listObjects()",
  "args": "[bucket, prefix, recursive, 1050 objects]",
  "duration": 30960,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = InternalError, message = We encountered an internal error, please try again.: cause(io: read/write on closed pipe), bucketName = minio-java-test-3228fjf, objectName = null, resource = /minio-java-test-3228fjf, requestId = 16809EA22287B389, hostId = a66e5883-3b74-448b-adc8-2e64ff40f6f7)\nrequest={method=GET, url=http://minio-c2.minio.io:32102/minio-java-test-3228fjf?list-type=2&prefix=minio&continuation-token=YldsdWFXOHRhbUYyWVMxMFpYTjBMWEV5Y210dlpWdHRhVzVwYjE5allXTm9aVHAyTVN4cFpEb3lOREJtTjJZMVpTMWhZVEprTFRRMFkyVXRZalkwWWkwMVlqQmlOalF3TTJaaU9UVmQ%3D&max-keys=1000&encoding-type=url&delimiter=, headers=Host: minio-c2.minio.io:32102\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210520T002445Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210520/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=500, headers=Accept-Ranges: bytes\nContent-Length: 373\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16809EA22287B389\nX-Xss-Protection: 1; mode=block\nDate: Thu, 20 May 2021 00:24:50 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.listObjectsV2(MinioClient.java:4368), io.minio.MinioClient$2$1.populateResult(MinioClient.java:2016), io.minio.MinioClient$ObjectIterator.populate(MinioClient.java:1882), io.minio.MinioClient$ObjectIterator.hasNext(MinioClient.java:1924), FunctionalTest.testListObjects(FunctionalTest.java:1398), FunctionalTest.listObjects(FunctionalTest.java:1455), FunctionalTest.runObjectTests(FunctionalTest.java:3747), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(10/15) Running minio-js tests ... done in 45 seconds
(11/15) Running minio-py tests ... FAILED in 1 minutes and 59 seconds
{
  "name": "minio-py:test_list_objects_with_1001_files",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-80fdd1a0-1593-4d74-980e-68940d075681",
    "object_name": "7c903d36-eb4c-4270-b7a1-1324b60f89eb_0 ~ 7c903d36-eb4c-4270-b7a1-1324b60f89eb_1000"
  },
  "message": "HTTPConnectionPool(host='minio-c2.minio.io', port=32102): Max retries exceeded with url: /minio-py-test-80fdd1a0-1593-4d74-980e-68940d075681?continuation-token=TjJNNU1ETmtNell0WldJMFl5MDBNamN3TFdJM1lURXRNVE15TkdJMk1HWTRPV1ZpWHpFNE9UaGJiV2x1YVc5ZlkyRmphR1U2ZGpFc2FXUTZPRFV3TkdSallqY3ROMlpsTmkwME1XUTBMVGxoTlRBdE1UZGhZVGszWVRkbVpUa3pYUT09&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=TjJNNU1ETmtNell0WldJMFl5MDBNamN3TFdJM1lURXRNVE15TkdJMk1HWTRPV1ZpWHpFNE9UaGJiV2x1YVc5ZlkyRmphR1U2ZGpFc2FXUTZPRFV3TkdSallqY3ROMlpsTmkwME1XUTBMVGxoTlRBdE1UZGhZVGszWVRkbVpUa3pYUT09 (Caused by ResponseError('too many 500 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1243, in test_list_objects_with_1001_files\n    _test_list_objects_api(bucket_name, no_of_created_files)\n  File \"/mint/run/core/minio-py/tests.py\", line 1123, in _test_list_objects_api\n    for obj in objects:\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2758, in _list_objects\n    response = self._execute(\"GET\", bucket_name, query_params=query)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 278, in _url_open\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 375, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 859, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 836, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 573, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-c2.minio.io', port=32102): Max retries exceeded with url: /minio-py-test-80fdd1a0-1593-4d74-980e-68940d075681?continuation-token=TjJNNU1ETmtNell0WldJMFl5MDBNamN3TFdJM1lURXRNVE15TkdJMk1HWTRPV1ZpWHpFNE9UaGJiV2x1YVc5ZlkyRmphR1U2ZGpFc2FXUTZPRFV3TkdSallqY3ROMlpsTmkwME1XUTBMVGxoTlRBdE1UZGhZVGszWVRkbVpUa3pYUT09&delimiter=%2F&list-type=2&max-keys=1000&prefix=&start-after=TjJNNU1ETmtNell0WldJMFl5MDBNamN3TFdJM1lURXRNVE15TkdJMk1HWTRPV1ZpWHpFNE9UaGJiV2x1YVc5ZlkyRmphR1U2ZGpFc2FXUTZPRFV3TkdSallqY3ROMlpsTmkwME1XUTBMVGxoTlRBdE1UZGhZVGszWVRkbVpUa3pYUT09 (Caused by ResponseError('too many 500 error responses',))\n",
  "duration": 79998
}
(11/15) Running s3cmd tests ... done in 1 minutes and 7 seconds
(12/15) Running s3select tests ... done in 3 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/data-scanner.go
+++ b/cmd/data-scanner.go
@@ -149,7 +149,6 @@ func runDataScanner(pctx context.Context, objAPI ObjectLayer) {
 			bf, err := globalNotificationSys.updateBloomFilter(ctx, nextBloomCycle)
 			logger.LogIf(ctx, err)
 			err = objAPI.NSScanner(ctx, bf, results)
-			close(results)
 			logger.LogIf(ctx, err)
 			if err == nil {
 				// Store new cycle...

--- a/cmd/erasure-server-pool.go
+++ b/cmd/erasure-server-pool.go
@@ -408,6 +408,9 @@ func (z *erasureServerPools) StorageInfo(ctx context.Context) (StorageInfo, []er
 }
 
 func (z *erasureServerPools) NSScanner(ctx context.Context, bf *bloomFilter, updates chan<- madmin.DataUsageInfo) error {
+	// Updates must be closed before we return.
+	defer close(updates)
+
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 

--- a/cmd/iam-object-store.go
+++ b/cmd/iam-object-store.go
@@ -19,7 +19,6 @@ package cmd
 
 import (
 	"context"
-	"errors"
 	"path"
 	"strings"
 	"sync"
@@ -76,12 +75,8 @@ func (iamOS *IAMObjectStore) runlock() {
 // location.
 //
 // 3. Migrate user identity json file to include version info.
-func (iamOS *IAMObjectStore) migrateUsersConfigToV1(ctx context.Context, isSTS bool) error {
+func (iamOS *IAMObjectStore) migrateUsersConfigToV1(ctx context.Context) error {
 	basePrefix := iamConfigUsersPrefix
-	if isSTS {
-		basePrefix = iamConfigSTSPrefix
-	}
-
 	for item := range listIAMConfigItems(ctx, iamOS.objAPI, basePrefix) {
 		if item.Err != nil {
 			return item.Err
@@ -110,9 +105,6 @@ func (iamOS *IAMObjectStore) migrateUsersConfigToV1(ctx context.Context, isSTS b
 			// 2. copy policy file to new location.
 			mp := newMappedPolicy(policyName)
 			userType := regularUser
-			if isSTS {
-				userType = stsUser
-			}
 			if err := iamOS.saveMappedPolicy(ctx, user, userType, false, mp); err != nil {
 				return err
 			}
@@ -124,7 +116,9 @@ func (iamOS *IAMObjectStore) migrateUsersConfigToV1(ctx context.Context, isSTS b
 	next:
 		// 4. check if user identity has old format.
 		identityPath := pathJoin(basePrefix, user, iamIdentityFile)
-		var cred auth.Credentials
+		var cred = auth.Credentials{
+			AccessKey: user,
+		}
 		if err := iamOS.loadIAMConfig(ctx, &cred, identityPath); err != nil {
 			switch err {
 			case errConfigNotFound:
@@ -138,15 +132,11 @@ func (iamOS *IAMObjectStore) migrateUsersConfigToV1(ctx context.Context, isSTS b
 		// If the file is already in the new format,
 		// then the parsed auth.Credentials will have
 		// the zero value for the struct.
-		var zeroCred auth.Credentials
-		if cred.Equal(zeroCred) {
+		if !cred.IsValid() {
 			// nothing to do
 			continue
 		}
 
-		// Found a id file in old format. Copy value
-		// into new format and save it.
-		cred.AccessKey = user
 		u := newUserIdentity(cred)
 		if err := iamOS.saveIAMConfig(ctx, u, identityPath); err != nil {
 			logger.LogIf(ctx, err)
@@ -171,26 +161,18 @@ func (iamOS *IAMObjectStore) migrateToV1(ctx context.Context) error {
 			// if IAM format
 			return err
 		}
-	} else {
-		if iamFmt.Version >= iamFormatVersion1 {
-			// Nothing to do.
-			return nil
-		}
-		// This case should not happen
-		// (i.e. Version is 0 or negative.)
-		return errors.New("got an invalid IAM format version")
 	}
 
-	// Migrate long-term users
-	if err := iamOS.migrateUsersConfigToV1(ctx, false); err != nil {
+	if iamFmt.Version >= iamFormatVersion1 {
+		// Nothing to do.
+		return nil
+	}
+
+	if err := iamOS.migrateUsersConfigToV1(ctx); err != nil {
 		logger.LogIf(ctx, err)
 		return err
 	}
-	// Migrate STS users
-	if err := iamOS.migrateUsersConfigToV1(ctx, true); err != nil {
-		logger.LogIf(ctx, err)
-		return err
-	}
+
 	// Save iam format to version 1.
 	if err := iamOS.saveIAMConfig(ctx, newIAMFormatVersion1(), path); err != nil {
 		logger.LogIf(ctx, err)

